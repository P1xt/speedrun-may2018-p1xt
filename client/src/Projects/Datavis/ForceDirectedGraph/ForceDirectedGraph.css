.ForceDirectedGraph {
  text-align: left;
  margin-top: 40px; }
  .ForceDirectedGraph > h1 {
    text-transform: none; }
