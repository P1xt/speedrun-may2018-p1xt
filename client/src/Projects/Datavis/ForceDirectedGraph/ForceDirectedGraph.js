import React from "react";
import { Container, Row, Col } from 'reactstrap';

import "./ForceDirectedGraph.css";

const ForceDirectedGraph = () => {
  return (
    <div className="ForceDirectedGraph">
      <Container>
        <Row>
          <Col>
            <h1>Force Directed Graph</h1>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default ForceDirectedGraph;
