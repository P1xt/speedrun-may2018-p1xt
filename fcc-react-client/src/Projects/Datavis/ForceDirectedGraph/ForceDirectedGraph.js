import React, { Component } from "react";
import { Card, CardBody } from "reactstrap";
import dataset from "./countries.json";
import * as d3 from "d3";
import d3Tip from "d3-tip";
import * as d3ScaleChromatic from "d3-scale-chromatic";

import "./ForceDirectedGraph.css";

class ForceDirectedGraph extends Component {
  constructor() {
    super();
    this.state = {
      data: null
    };
  }
  /**
   * componentWillMount
   * Note, it's important to pull data in from json
   * file in componentWillMount so that it's available
   * on render
   * @return {void}
   * @memberof ForceDirectedGraph
   */
  componentWillMount() {
    const data = JSON.parse(JSON.stringify(dataset));
    this.setState({
      data
    });
  }

  /**
   *componentDidMount
   * Note, it's important to build the chart
   * after componentWillMount run to ensure the
   * data is available. So, build the chart here
   * in componentDidMount
   * @return {void}
   * @memberof ForceDirectedGraph
   */
  componentDidMount() {
    this.buildChart();
  }

  /**
   * responsiveChart
   * Turns the selected chart responsive
   *
   * Adapted from:
   * https://brendansudol.com/writing/responsive-d3
   * @memberof ForceDirectedGraph
   */
  responsiveChart = svg => {
    const container = d3.select(svg.node().parentNode),
      width = parseInt(svg.style("width"), 10),
      height = parseInt(svg.style("height"), 10),
      aspect = width / height;

    const resize = () => {
      const targetWidth = parseInt(container.style("width"), 10);
      svg.attr("width", targetWidth);
      svg.attr("height", Math.round(targetWidth / aspect));
    };

    svg
      .attr("viewBox", "0 0 " + width + " " + height)
      .attr("perserveAspectRatio", "xMinYMid")
      .call(resize);

    d3.select(window).on("resize." + container.attr("id"), resize);
  };

  buildChart() {}

  render() {
    return (
      <div className="ForceDirectedGraph">
        <div className="wrapper">
          <Card>
            <CardBody>
              <div id="chart" />
            </CardBody>
          </Card>
        </div>
      </div>
    );
  }
}

export default ForceDirectedGraph;
