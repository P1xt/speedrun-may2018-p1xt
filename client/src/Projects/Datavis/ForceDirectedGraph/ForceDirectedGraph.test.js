import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import ForceDirectedGraph from "./ForceDirectedGraph";

describe("<ForceDirectedGraph />", () => {
  it("renders without crashing", () => {
    shallow(<ForceDirectedGraph />);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<ForceDirectedGraph />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
