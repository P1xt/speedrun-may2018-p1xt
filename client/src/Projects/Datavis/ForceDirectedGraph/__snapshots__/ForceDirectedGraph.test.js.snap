// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForceDirectedGraph /> renders correctly 1`] = `
<div
  className="ForceDirectedGraph"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Force Directed Graph
        </h1>
      </div>
    </div>
  </div>
</div>
`;
